src/rabbit_dead_letter.erl

```diff
@@ -84,36 +84,85 @@ x_death_event_key(Info, Key, KeyType) ->
         {value, {Key, KeyType, Val}} -> Val
     end.
 
+maybe_append_to_event_group(Info, _Queue, _Reason, []) ->
+    [Info];
+maybe_append_to_event_group(Info, Queue, Reason, Acc) ->
+    case lists:any(queue_and_reason_matcher(Queue, Reason), Acc) of
+        true  -> Acc;
+        false -> [Info | Acc]
+    end.
+
+group_by_queue_and_reason(Tables) ->
+    Infos = [Info || {table, Info} <- Tables],
+    Grouped =
+        lists:foldl(fun (Info, Acc) ->
```

Comment:
As the other comment down below, here you can push the `fun` to the next line and then just use Emacs indent like we do here for example https://github.com/rabbitmq/rabbitmq-server/blob/master/src/rabbit_msg_store.erl#L2019